Add slaking documents (#422)
hpcugent/vsc_user_docs: intro-HPC/ch_interactive_gent.tex

> \chapter{HPC-UGent interactive cluster}

Reviewer: `debug` should also be in chapter title imho

Reviewer 2: i don't agree. we really should try to sell it for its interactivity

Reviewer: fine, whatever, just saying that "interactive cluster" is just wrong, we're not forcing only interactive jobs in any way, so it will filled up with short batch jobs in no time

\label{ch:interactive_ugent}

\section{Purpose}
\label{sec:interactive_ugent_pupose}
The purpose of this cluster is to give the user an environment where
there should be no waiting in the queue to get access to a limited
number of resources. This environment allows a user to immediatelty

> start doing, and is the ideal place for interactive work such as

Author: working?

> development, debugging and light prodcution (typically sufficient

Reviewer: `prodcution` -> `production workloads`

> for training and/or courses). This enviroment should be seen as an

Reviewer: `This environtment should ... ` => start new paragraph here

> extension of the loginnodes, instead of the a compute resource.

Reviewer: `loginnodes` -> `login nodes`

last part doesn't make sense: `of the a` (missing word?)

> The interactive cluster is overcommitted, which means that more CPU cores can be

Reviewer: `overcommitted` in bold or `\emph{overcommitted}`?

> assigned to jobs than physically exist in the cluster. Obviously, the performance of this cluster

Reviewer: `assigned` -> `requested`?

heavily depends on the workloads and that actual overcommit usage. Be aware that jobs can slow
down or speed up during their execution.

> \\ Due to the restrictions and sharing the CPU resources (see section~\ref{subsec:interactive_ugent_restrictions}) 

Reviewer: just leave empty line before rather than using `\\`?

Reviewer: `shared of the CPU resources` (missing `of`)

jobs on slaking should normally start more or less immediately. As a consequence,
typical workloads for this cluster should be limited to:
\begin{itemize}

>   \item  Run interactive jobs (see chapter~\ref{ch:running-interactive-jobs}) with lots of idle time

Reviewer: `Run interactive jobs` -> `Interactive jobs`

Reviewer 2: i would also remove the "with lots of idle time". we stress enough that we trade interactivty for performance, and if we don't repeat it a few more times ;)

         (courses, hands-ons, etc.) 

>   \item  Run Cluster Desktops (see chapter~\ref{ch:web_portal}) with lots of idle time

Reviewer: `Run Cluster Desktop` -> `Cluster desktop sessions`

         (courses, hands-ons, etc.)

>   \item  Run small jobs for which performance is not an issue (courses, hands-ons, etc.)

Reviewer: `Run small jobs` -> `Small jobs`

  \item  Debugging programs

>   \item  Try/debug submit scripts

Reviewer: `Tru/debug submit script` -> `Testing and debugging of job scripts`

\end{itemize} 

\section{Submitting jobs}
\label{sec:interactive_ugent_jobs}

> To submit jobs to the HPC-UGent interactive cluster nicknamed \lstinline|slaking|, first use:

Reviewer: also mention `debug`?

\begin{prompt}
%\shellcmd{module swap cluster/slaking}
\end{prompt}

> Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc. commands.

Reviewer: refer to general section on submitting jobs?

\subsection{Restrictions and overcommit factor}
\label{subsec:interactive_ugent_restrictions}

> Currently each user may have at most 5 jobs in the queue (both running and waiting to run)

Reviewer: drop `Currently`

and at most 3 jobs per user can be running at the same time. The running jobs may allocate 

> no more than 8 CPU cores in total. \\

Reviewer: drop `\\`, just add empty line below

> no more than 8 CPU cores in total. \\

Reviewer: Make make this a 3-item list, to make it stand out a bit more?

```
Some per-user limits are in place for this cluster:
\begin{itemize}
\item each user may have at most 5 jobs in the queue (both running and waiting to run);
\item at most 3 jobs per user can be running at the same time;
\item running jobs  may allocate no more than 8 CPU cores in total, per user;
\end{itemize}

Reviewer: `Per-user limits` is enough (implies that it's across all jobs of that user, `total` is probably confusing)

> At this moment, the cluster has an overcommit factor 6. This means that 6 times more cores

Reviewer: Drop `At this moment`:

```
In addition, the cluster has an overcommit factor of 6.
```

(missing `of`)

can be allocated than physically exist. Simulatenously, the default memory per core is 6
times less than what would physically be available on a non-overcommitted cluster. 

> As a first sight the overcommit ratio might seen high, but this value is rather typical

Reviewer 3: ```suggestion
At first sight the overcommit ratio might seen high, but this value is rather typical
```

> for cloud resources that not dedicated to compute. \\

Reviewer 4: that **are** not

Author: fixed

> for cloud resources that are not dedicated to compute. \\

Reviewer: drop this sentence, overkill imho, we don't need to defend our configuration in docs, just explain it

> Please note that based on the (historical) workload of the interactive cluster, the above

Reviewer: empty line above (to make it stand out a bit more, since this also applies to the per-user limits, not only the overcommit part)

\label{ch:interactive_ugent}

\section{Purpose}
\label{sec:interactive_ugent_pupose}
The purpose of this cluster is to give the user an environment where
there should be no waiting in the queue to get access to a limited
number of resources. This environment allows a user to immediatelty
start working, and is the ideal place for interactive work such as
development, debugging and light prodcution workloads (typically sufficient
for training and/or courses). \\ This enviroment should be seen as an
extension of the login nodes, instead of a dedicated compute resource.
The interactive cluster is \emph{overcommitted}, which means that more CPU cores can be
requested to jobs than physically exist in the cluster. Obviously, the performance of this cluster
heavily depends on the workloads and the actual overcommit usage. Be aware that jobs can slow
down or speed up during their execution.
 \\
Due to the restrictions and sharing of the CPU resources (see section~\ref{subsec:interactive_ugent_restrictions}) 
jobs on slaking should normally start more or less immediately. As a consequence,
typical workloads for this cluster should be limited to:
\begin{itemize}

>   \item  Interactive jobs (see chapter~\ref{ch:running-interactive-jobs}) with lots of idle time

Reviewer 2: remove the `(courses, hands-ons, etc.)` in first 3 items; and remove "with lots of idle time" in first 2.

         (courses, hands-ons, etc.) 
  \item  Cluster desktop sessions (see chapter~\ref{ch:web_portal}) with lots of idle time
         (courses, hands-ons, etc.)

>   \item  Small jobs for which performance is not an issue (courses, hands-ons, etc.)

Reviewer 2: `Small jobs for which performance is not an issue (courses, hands-ons, etc.)` -> `Light jobs`? we should stress the "performance is not an issue" elsewhere

Reviewer 2: ok

  \item  Debugging programs
  \item  Testing and debuging submit scripts
\end{itemize} 

\section{Submitting jobs}
\label{sec:interactive_ugent_jobs}

To submit jobs to the HPC-UGent interactive and debug cluster nicknamed \lstinline|slaking|, first use:

\begin{prompt}
%\shellcmd{module swap cluster/slaking}
\end{prompt}

Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc. commands. (see chapter~\ref{ch:running-batch-jobs})

\subsection{Restrictions and overcommit factor}
\label{subsec:interactive_ugent_restrictions}

Some per-user limits are in place for this cluster:
\begin{itemize}
  \item each user may have at most 5 jobs in the queue (both running and waiting to run);
  \item at most 3 jobs per user can be running at the same time;
  \item running jobs may allocate in total no more than 8 CPU cores;
\end{itemize}

In addition, the cluster has an overcommit factor of 6. This means that 6 times more cores

> can be allocated than physically exist. Simultaneously, the default memory per core is 6

Reviewer 3: there are now.
